src/components/application_manager/src/commands/mobile/set_app_icon_request.cc

```diff
@@ -190,31 +190,41 @@ void SetAppIconRequest::CopyToIconStorage(
 void SetAppIconRequest::RemoveOldestIcons(const std::string& storage,
                                           const uint32_t icons_amount) const {
   const std::vector<std::string> icons_list = file_system::ListFiles(storage);
-  std::map<uint64_t, std::string> icon_modification_time;
+  std::vector<file_system::FileDescriptor> info_about_icons;
```

Comment:
@dcherniev Of course you can, but you have to change slightly your code and wrap already implemented function in a functor.